Added datediff calculation in spark

TomWhite-MedStar opened the pull request:
Added  translations and test cases for Spark datediff() for non-day intervals.
Solves issue #345

schuemie commented:
Thanks! Two questions:

- Do we still need rules for other dialects? (E.g. Snowflake)
- Just curious what your thoughts are for rules that don't actually tanslate (e.g. `redshift,"DATEADD(day,@days,@date)","DATEADD(day,@days,@date)"`

TomWhite-MedStar commented:
I don't have access to test environments for the other languages.  Who can be engaged to make the edits for those?

I am ambivalent about rules that don't translate.  One option is to keep them there so that authors know they were considered, unless it risks getting the parser confused.

schuemie commented:
Let's work on the other dialects at the hack-a-thon.

I'm leaning towards not including rules that don't translate. There are many things that don't need translation. If we add all those the set of rules will become massive. It is also a bit of an open set. For example, we don't translate `SELECT`. Should we add non-translation rules for that?

schuemie commented:
Sorry, this one slipped of my radar. Would you mind removing the rules that don't translate?

TomWhite-MedStar commented:
@schuemie , I removed the translations that didn't do anything.
I also added translations for negative sub-day time intervals to support the work on https://github.com/OHDSI/Atlas/issues/2886

TomWhite-MedStar commented:
@schuemie - Note, I only added time-based translations for hour, minute and second.  I did not change the  default translations for days.  This should keep the default behavior (e.g. if add days to a datetime value, the logic will effectively strip off the time portion first).    If users truly want 24 hour periods, as opposed to calendar days, they can add 24 * N hours, instead of N days.
